packages/components/collapse/src/Collapse.tsx

```diff
@@ -46,30 +48,26 @@ export const Collapse = ({
 
     const handleTransitionEnd = () => {
       if (current) {
-        current.style.height = 'auto';
+        if (isExpanded) {
+          current.style.removeProperty('height');
+        } else {
+          current.style.setProperty('display', 'none');
+        }
       }
     };
 
     if (current) {
-      if (isExpanded) {
-        current.addEventListener('transitionend', handleTransitionEnd);
-
-        requestAnimationFrame(function () {
-          current.style.height = '0px';
+      current.addEventListener('transitionend', handleTransitionEnd);
+      const fromHeight = isExpanded ? '0px' : getPanelContentHeight();
+      const toHeight = isExpanded ? getPanelContentHeight() : '0px';
+      requestAnimationFrame(function () {
+        current.style.removeProperty('display');
```

Comment:
We have to remove property `display: none` before we get PanelContentHeight, because otherwise, it will be always `0`

Reply:
I'm moving it below so the display is reset to the default before calculating the height. We don't need to wait for another animation frame to do so, do we? I'm not sure but it worked for me without waiting for another frame.